migrate.py

```diff
@@ -389,6 +394,12 @@ def dir_to_path(path):
                     ptype, ns, coll,
                 )
 
+    dupes = {k: v for k, v in files_to_collections.items() if len(v) > 1}
+    if dupes:
```

Comment:
I think it's nicer to log things like this:
```suggestion
    dup_errors = []
    for file_path, coll_names in dupes.items():
        err_msg = f'{file_path} is assigned to the following collections: {", ".join(coll_names)}'
        logger.error(err_msg)
        dup_errors.append(err_msg)
    if dup_errors:
        raise RuntimeError(
            'Each plugin needs to be assigned to one collection only. '
            'The following files are assigned to multiple collections: %s.',
            "; ".join(dup_errors),
        )
```